packages/mdc-floating-label/_mixins.scss

```diff
@@ -21,60 +21,148 @@
-  @include mdc-theme-prop(background-color, $color);
-}
+  // postcss-bem-linter: define floating-label
+  .mdc-floating-label {
+    @include mdc-typography(subtitle1, $query: $query);
 
-@mixin mdc-floating-label-shake-keyframes($modifier, $positionY, $positionX: 0%, $scale: .75) {
-  @keyframes mdc-floating-label-shake-float-above-#{$modifier} {
-    0% {
-      transform: translateX(calc(0 - #{$positionX})) translateY(-#{$positionY}) scale(#{$scale});
+    @include mdc-feature-targets($feat-structure) {
+      position: absolute;
+      /* @noflip */
+      left: 0;
+      /* @noflip */
+      transform-origin: left top;
+      line-height: 1.15rem;
+      text-align: left;
+      text-overflow: ellipsis;
+      white-space: nowrap;
+      cursor: text;
+      overflow: hidden;
+
+      @include mdc-rtl {
+        /* @noflip */
+        right: 0;
+        /* @noflip */
+        left: auto;
+        /* @noflip */
+        transform-origin: right top;
+        /* @noflip */
+        text-align: right;
+      }
     }
 
-    33% {
-      animation-timing-function: cubic-bezier(.5, 0, .701732, .495819);
-      transform: translateX(calc(4% - #{$positionX})) translateY(-#{$positionY}) scale(#{$scale});
+    @include mdc-feature-targets($feat-animation) {
+      transition:
+        transform $mdc-floating-label-transition-duration $mdc-animation-standard-curve-timing-function,
+        color $mdc-floating-label-transition-duration $mdc-animation-standard-curve-timing-function;
+      /* @alternate */
+      // Force the label into its own layer to prevent visible layer promotion adjustments
+      // when the ripple is activated behind it.
+      will-change: transform;
     }
+  }
 
-    66% {
-      animation-timing-function: cubic-bezier(.302435, .381352, .55, .956352);
-      transform: translateX(calc(-4% - #{$positionX})) translateY(-#{$positionY}) scale(#{$scale});
+  .mdc-floating-label--float-above {
+    @include mdc-feature-targets($feat-structure) {
+      cursor: auto;
     }
+  }
+
+  @at-root {
+    @include mdc-floating-label-float-position($mdc-floating-label-position-y, $query: $query);
+    @include mdc-floating-label-shake-animation(standard, $query: $query);
+  }
+
+  @include mdc-floating-label-shake-keyframes(standard, $mdc-floating-label-position-y, $query: $query);
+}
+
+@mixin mdc-floating-label-ink-color($color, $query: mdc-feature-all()) {
+  $feat-color: mdc-feature-create-target($query, color);
+
+  @include mdc-feature-targets($feat-color) {
+    @include mdc-theme-prop(color, $color);
+  }
+}
+
+// Used for textarea in case of scrolling
+@mixin mdc-floating-label-fill-color($color, $query: mdc-feature-all()) {
+  $feat-color: mdc-feature-create-target($query, color);
+
+  @include mdc-feature-targets($feat-color) {
+    @include mdc-theme-prop(background-color, $color);
+  }
+}
+
+@mixin mdc-floating-label-shake-keyframes($modifier, $positionY, $positionX: 0%, $scale: .75,
+$query: mdc-feature-all()) {
```

Comment:
The indentation here is weird because the top line was over the limit so I had to break it up, but stylelint starts complaining if I add any indentation to it.